code/game/objects/items/weapons/shields.dm

```diff
@@ -68,30 +143,53 @@
 	throw_range = 4
 	w_class = WEIGHT_CLASS_BULKY
 	origin_tech = list(TECH_MATERIAL = 2)
-	matter = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_GLASS = 7500)
+	matter = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_PLASTIC = 7500)
+	material_repair_type = MATERIAL_PLASTIC
 	attack_verb = list("shoved", "bashed")
-	var/cooldown = 0 //shield bash cooldown. based on world.time
 
-/obj/item/shield/riot/handle_shield(mob/user)
+	armor = list(
+		melee = ARMOR_MELEE_VERY_HIGH,
+		bullet = ARMOR_BALLISTIC_MINOR,
+		laser = ARMOR_LASER_MINOR,
+		energy = ARMOR_ENERGY_MINOR,
+		bomb = ARMOR_BOMB_PADDED
+	)
+
+	/// shield bash cooldown. based on world.time
+	var/cooldown = 0
+
+/obj/item/shield/riot/update_icon()
+	if(shield_health == 0)
+		icon_state = "[initial(icon_state)]_broken"
+		item_state = "[initial(icon_state)]_broken"
+	else
+		icon_state = "[initial(icon_state)]"
+		item_state = "[initial(icon_state)]"
+
+	if(ismob(loc))
+		var/mob/user = loc
+		user.update_inv_l_hand()
+		user.update_inv_r_hand()
+		user.update_inv_back()
+
+/obj/item/shield/riot/handle_damage(mob/user, damage)
 	. = ..()
-	if(.) playsound(user.loc, 'sound/weapons/Genhit.ogg', 50, 1)
+	if(shield_health == 0)
+		update_icon()
 
-/obj/item/shield/riot/get_block_chance(mob/user, var/damage, atom/damage_source = null, mob/attacker = null)
-	if(istype(damage_source, /obj/projectile))
-		var/obj/projectile/P = damage_source
-		//plastic shields do not stop bullets or lasers, even in space. Will block beanbags, rubber bullets, and stunshots just fine though.
-		if((is_sharp(P) && damage > 10) || istype(P, /obj/projectile/beam))
-			return 0
-	return base_block_chance
+/obj/item/shield/riot/handle_shield(mob/user)
+	. = ..()
+	if(.)
+		playsound(user.loc, 'sound/weapons/Genhit.ogg', 50, 1)
```

Comment:
```suggestion
		playsound(get_turf(user), 'sound/weapons/Genhit.ogg', 50, 1)
```
Extrarange needs to be one of the defines, or based on one of the defines, no raw numbers